### PR TITLE
Feature stock from db

### PR DESCRIPTION
**Pull request by ronp2001**
Functionality to create update(not fully just updating the list from what is already in the db) and read observables in the data-storage service class. so that the stock screen can subscribe to these lists to check for changes and so can any other screen that might need this data

### CODE REVIEW

File: src/app/stock/stock.component.ts [hunk 1]
```diff
@@ -1,88 +1,44 @@
...
-    new Packaging('dozen', '1', 2, 1, 'Rode Doos27', 'Rotterdam'),
-    new Packaging('dozen', '2', 20, 10, 'Blauwe Doos28', 'Amsterdam'),
-    new Packaging('dozen', '1', 2, 1, 'Rode Doos29', 'Rotterdam'),
-    new Packaging('dozen', '2', 20, 10, 'Blauwe Doos30', 'Amsterdam'),
-    new Packaging('dozen', '1', 2, 1, 'Rode Doos31', 'Rotterdam'),
-    new Packaging('dozen', '2', 20, 10, 'Blauwe Doos32', 'Amsterdam'),
-    new Packaging('dozen', '1', 2, 1, 'Rode Doos33', 'Rotterdam'),
-    new Packaging('dozen', '2', 20, 10, 'Blauwe Doos34', 'Amsterdam'),
-  ];
+  private unsubscribe$ = new Subject<void>();
```
Comment (davidvdw02) [inline src/app/stock/stock.component.ts:24]: unsubscribe beetje onduidelijke naam imo

File: src/app/stock/stock.component.ts [hunk 2]
```diff
@@ -1,88 +1,44 @@
...
 
-  mockLocations: string[] = [
-    'Rotterdam', 'Amsterdam', 'Brabant'
-  ];
-
-  constructor(private http: DataStorageService){}
+  constructor(private http: DataStorageService) {}
```
Comment (davidvdw02) [inline src/app/stock/stock.component.ts:26]: dit is niet de http service.
```suggestion
  constructor(private dataStorageService: DataStorageService) {}
```

File: src/app/stock/stock.component.ts [hunk 3]
```diff
@@ -100,22 +56,57 @@ export class StockComponent {
...
+    this.http.locationNames$
+      .pipe(takeUntil(this.unsubscribe$))
+      .subscribe(locationNames => {
+        this.locationNames = locationNames;
+      })
+  }
+   
+  ngOnDestroy() {
+    this.unsubscribe$.next();
+    this.unsubscribe$.complete();
```
Comment (davidvdw02) [inline src/app/stock/stock.component.ts:110]:
```suggestion
  this.unsubscribe$.unsubscribe();
```

File: src/app/stock/add-package-popup/add-package-popup.component.ts [hunk 4]
```diff
@@ -23,27 +22,23 @@ export class AddPackagePopupComponent {
...
-      this.newPackage.get('amount')?.value, 
-      this.newPackage.get('minAmount')?.value, 
-      this.newPackage.get('name')?.value, 
-      'TestLocation');
-
-    if(packaging === undefined){
-        
-    } else if(this.checkNewPackage(packaging)){
-      if(this.checkAmount(packaging)){
+    if (packaging === undefined) {
```
[the remark this answers is not in the document]
Comment (ronp2001) [reply src/app/stock/add-package-popup/add-package-popup.component.ts:36]: @DaaromThomas

File: src/app/stock/add-package-popup/add-package-popup.component.ts [hunk 5]
```diff
@@ -57,41 +52,40 @@ export class AddPackagePopupComponent {
...
-    const minAmount: number = Number(packaging.getMinAmount);
+    const amount: number = Number(packaging.amountinstock);
+    const minAmount: number = Number(packaging.minAmount);
     if (isNaN(amount)) {
-        this.error = "Amount is not a number";
-        return true;
-    }else if(isNaN(minAmount)){
-      this.error = "Minimal amount is not a number";
+      this.error = 'Amount is not a number';
+      return true;
```
[the remark this answers is not in the document]
Comment (ronp2001) [reply src/app/stock/add-package-popup/add-package-popup.component.ts:83]: @DaaromThomas

File: src/app/stock/add-package-popup/add-package-popup.component.ts [hunk 6]
```diff
@@ -23,27 +22,23 @@ export class AddPackagePopupComponent {
...
+  constructor(private storageService: DataStorageService) {}
+
   done(): void {
     this.amountErrorHidden = true;
+    const packaging: Packaging = {
+      packagingGroup: this.newPackage.value.group,
+      amountinstock: this.newPackage.value.amount,
+      minAmount: this.newPackage.value.minAmount,
+      name: this.newPackage.value.name
+    };
```
[the remark this answers is not in the document]
Comment (ronp2001) [reply src/app/stock/add-package-popup/add-package-popup.component.ts:34]: @DaaromThomas

File: src/app/stock/stock.component.ts (hunk 3 above)
Comment (ronp2001) [reply src/app/stock/stock.component.ts]: see commit b4bf2bc

File: src/app/stock/stock.component.ts (hunk 2 above)
Comment (ronp2001) [reply src/app/stock/stock.component.ts]: see commit b4bf2bc

File: src/app/stock/stock.component.ts (hunk 1 above)
Comment (ronp2001) [reply src/app/stock/stock.component.ts]: see commit b4bf2bc

File: src/app/stock/stock.component.ts [hunk 7]
```diff
@@ -1,88 +1,44 @@
 // stock.component.ts
```
[the remark this answers is not in the document]
Comment (ronp2001) [reply src/app/stock/stock.component.ts]: see commit b4bf2bc

File: src/app/services/data-storage.service.ts [hunk 8]
```diff
@@ -1,33 +1,70 @@
...
+import { Subject, tap, forkJoin } from 'rxjs';
+import { Stock } from '../interfaces/stock';
 
 @Injectable({
-    providedIn: 'root'
+  providedIn: 'root',
 })
 export class DataStorageService {
+  baseurl: string = 'http://localhost:8080';
+  packageList$: Subject<Packaging[]> = new Subject<Packaging[]>();
```
[the remark this answers is not in the document]
Comment (ronp2001) [reply src/app/services/data-storage.service.ts]: see commit b4bf2bc

**Comment by ronp2001** [comment]
Do not merge yet, found issue where package storePackage gets called twice, fixing it now (and trying to make something a bit more future proof in the mean time)

**Comment by ronp2001** [comment]
> Do not merge yet, found issue where package storePackage gets called twice, fixing it now (and trying to make something a bit more future proof in the mean time)

Issue should be fixed

**Review by DaaromThomas — APPROVED** [review]
Ziet er naar mijn idee goed uit

**Review by davidvdw02 — APPROVED** [review]

**Comment by davidvdw02** [comment]
mannen code

**Pull request merged**